learning_resources_search/serializers.py

```diff
@@ -372,9 +373,33 @@ class SearchResponseMetadata(TypedDict):
 
 class SearchResponseSerializer(serializers.Serializer):
     count = serializers.SerializerMethodField()
+    next = serializers.SerializerMethodField()
+    previous = serializers.SerializerMethodField()
     results = serializers.SerializerMethodField()
     metadata = serializers.SerializerMethodField()
 
+    def construct_pagination_url(self, instance, request, link_type="next"):
+        if request:
+            url = request.build_absolute_uri()
+            total_record_count = self.get_count(instance)
+            offset = int(request.query_params.get("offset", 0))
+            limit = int(request.query_params.get("limit", 20))
+            if link_type == "previous":
+                offset -= limit
+            else:
+                offset += limit
+            if offset >= 0 and offset <= total_record_count:
```

Comment:
Maybe make this `< total_record_count` instead of `<= total_record_count`?  I just happened to use a [query](http://localhost:8063/api/v1/learning_resources_search/?q=algebra) resulting in exactly 80 results, and while advancing through the next links, the last one (`http://localhost:8063/api/v1/learning_resources_search/?offset=80&q=algebra`) returned 0 results:

<img width="744" alt="Screenshot 2024-02-28 at 10 12 17 AM" src="https://github.com/mitodl/mit-open/assets/187676/94625598-6a07-40a1-bc3d-d4a3e2ed462a">

Reply:
Good catch. fixed.